### DISCUSSION TITLE
"An unexpected error has occured" upon page creation

### DISCUSSION THREAD

**Discussion by ghost** [requarks/wiki#3659] [Error / Bug Report]
**Describe the bug**
Every time a new page is created in markdown or WYSIWYG format, I get a small red box that appears at the top that reads "An unexpected error has occurred" this error message appears for a very brief period of time. The page then goes on to get created and nothing else appears to be out of place. The page renders properly. I've looked at the logs and no error appears.

**To Reproduce**
1. Create new page
2. Choose your editor
3. Enter content on the page
4. Click the "Create" button
5. Page renders
6. Error message gets displyed very briefly
7. Page reload to display the rendered content 

**Screenshots**
The error message:
![Error](https://user-images.githubusercontent.com/63517511/79053886-b964d180-7c0e-11ea-8882-85e83681130d.png)

**Host Info (please complete the following information):**
 - OS: Ubuntu 18.04, Docker
 - Wiki.js version: 2.2.51
 - Database engine: SQLite

**Comment by Smankusors** [comment]
well, I think there's one log you forgot, the browser console. Can you please check your browser's DevTools? See if there's any console output there.

**Comment by ghost** [comment]
Good idea. Here's what I found from the firefox console:

`[GraphQL error]: Message: This page does not exist., Location: [object Object], Path: pages,checkConflicts app.948c70dcebb94223ec54.js:2:2071`

`GraphQL execution errors for query 'isConflict' app.948c70dcebb94223ec54.js:36:54570`

`{…}
​
extensions: Object { code: "INTERNAL_SERVER_ERROR", exception: {…} }
​
locations: Array [ {…} ]
​
message: "This page does not exist."
​
path: Array [ "pages", "checkConflicts" ]
​
<prototype>: Object { … }
app.948c70dcebb94223ec54.js:36:54718`

`TypeError: e.pages is undefinededitor.89c74b8d96cfe6d6a3a9.js:2:13802`

**Comment by Smankusors** [comment]
is this happen with other browsers?

**Comment by ghost** [comment]
I have not seen it in any other browser

**Comment by Smankusors** [comment]
It could be only your browser problem I think. Can you try with private mode? See if this still happens?

---

Btw, what is your Firefox version?

**Comment by ghost** [comment]
I just tried in Private Mode and still get the error. I'm using firefox 74.0.1 on macOS 10.15.4. Here's a screenshot of the errors if that helps at all. Also, I'm not sure if it's related but when I create a new page by following a link to a page that doesn't exist yet, once the page gets created and I get this error, the original link does not change from red to blue to indicate that the page exists. I can click the link to get to the new page and it works but that link stays red.

![Error2](https://user-images.githubusercontent.com/63517511/79077835-21302080-7cd2-11ea-9489-8a9a85fef571.png)

**Comment by Smankusors** [comment]
I think this is related with your other issue #1708. I think somehow Wiki.js confused what current locale are. What is your locale should be? Arabic? Or else?

> could be corrupt database tho

**Comment by ghost** [comment]
My locale is whatever a normal locale in Canada should be, I'm assuming. How do I go about figuring that out?

Also, just for the heck of it, I deployed the docker container along with the postgresql container as per the example docker-compose file in the doc and I'm seeing the same behaviour. Same thing with the links not turning blue.

**Comment by ghost** [comment]
I just found how to make the links turn blue. Once the page has been created, saved and rendered, you go back and edit the page you just created and modify it. Once you saved it again, the link will turn from red to blue. I'm guessing it has something to do with the db somehow.

**Comment by DustinVenegas** [comment]
The same error is appearing with Firefox 75.0 (64-bit) Windows and wiki.js 2.3.81.

There's a single GraphQL request that being made that appears related. The ID property (`"id":0`) tells me the query parameters are suspect.

**Query**

```json
{
  "0": {
    "extensions": {},
    "operationName": null,
    "query": "query ($id: Int!, $checkoutDate: Date!) {\n  pages {\n    checkConflicts(id: $id, checkoutDate: $checkoutDate)\n    __typename\n  }\n}\n",
    "variables": {
      "checkoutDate": "2020-05-04T17:14:04.997Z",
      "id": 0
    }
  }
}
```
